client/comms/wsconn.go

```diff
@@ -61,6 +61,11 @@ type WsCfg struct {
 	// ReconnectSync runs the needed reconnection synchronization after
 	// a disconnect.
 	ReconnectSync func()
+	// ConnectEventFunc runs whenever connection status changes.
+	//
+	// NOTE: These notifications are for aesthetic purposes only. They
+	// should not be relied upon for operation.
```

Comment:
I find the wording "relied upon for operation" too vague.  This says to me don't do anything with the event data except display it.  But I that is not the only thing you can do with it.  The receiver getting a connected=false event can certainly make use of that, stopping new orders from being submitted or otherwise changing the errors.  While connected=true does NOT mean that we guarantee the link is up until a connect=false event (submitting an order on a broken connection could trigger the disconnected state), it does mean we think it's connected and it's OK to try submitting orders.

Reply:
To recap the discussion with rpcclient and dcrd notifications, the objection to a disconnect notification was that the consumer might believe that it was connected up until the instant of the notification OR believe that the absence of the notification means the connection was never lost.  That could be especially problematic with rpcclient's auto-reconnect that could hide the state changes from the consumer.  I don't think any of this applies to DEX client.  We are managing the connection, with all reconnect attempts triggering this event (when wsConn first discovers it is broken and then when keepAlive>connect restores it).